lib/elixir/lib/uri.ex

```diff
@@ -71,44 +71,67 @@ defmodule URI do
   end
 
   @doc """
-  Encodes an enumerable into a query string.
+  Encodes `enumerable` into a query string using `encoding`.
 
   Takes an enumerable that enumerates as a list of two-element
   tuples (for instance, a map or a keyword list) and returns a string
-  in the form of `key1=value1&key2=value2...` where keys and
-  values are URL encoded as per `encode_www_form/1`.
+  in the form of `key1=value1&key2=value2...`.
 
   Keys and values can be any term that implements the `String.Chars`
   protocol with the exception of lists, which are explicitly forbidden.
 
+  You can specify one of the following `encoding` strategies:
+
+    * `:www_form` - (default) keys and values are URL encoded as per
+      `encode_www_form/1`. This is the format typically used by browsers
+      on query strings and form data. It encodes " " as "+".
+
+    * `:rfc_3986` - the same as `:www_form` except it encodes " " as
```

Comment:
```suggestion
    * `:www_form` - (default, since v1.12.0) keys and values are URL encoded as per
      `encode_www_form/1`. This is the format typically used by browsers
      on query strings and form data. It encodes " " as "+".

    * `:rfc_3986` - (since v1.12.0) the same as `:www_form` except it encodes " " as
```